### ISSUE TITLE
Parser should ignore whitespace and set types

### ISSUE THREAD

**Issue by LynnCo** [theskumar/python-dotenv#7]
The parser should strip leading and trailing whitespace on KEY and "Value" and also set types for booleans `r'(?i)(true|false)` and integers `r'[0-9]*'`

Happy to submit a PR for this, mostly just wondering if you agree that these things should happen

**Comment by theskumar** [comment]
TLDR;
- type conversion - nope.
- striping - go for it.

---

__Automatic Type Conversion__

While this is useful in most cases, I am afraid this feature is under the scope of this project. The goal of this project is:

> Read the data from `.env` and export it as as environment variable. Once they are injected as environment variables, they can be read by any python code any where in your codebase, using `os.environ.get()`, etc. 

Things put in operating system's environment variable are always `string`.

There is an nice interface provided by https://github.com/joke2k/django-environ to read the available environment variables and then cast them to desired datatype. I would encourage you to have a look at it. 

__Striping whitespaces__

We can have it enabled via an optional parameter (`strip_whitespaces=True`) to `load_dotenv` function with default value as `False`. Happy to see a pull request. :blush: :+1:

**Comment by josephmancuso** [comment]
@theskumar is automatic type casting still not within scope?
